src/abstractdataframe/abstractdataframe.jl

```diff
@@ -946,7 +947,7 @@ function _vcat(dfs::AbstractVector{<:AbstractDataFrame})
     for (i, name) in enumerate(header)
         data = [df[name] for df in dfs]
         lens = map(length, data)
-        cols[i] = promote_col_type(data...)(undef, sum(lens))
+        cols[i] = promote_col_type(Tuple(data))(undef, sum(lens))
```

Comment:
I'd just pass `data` directly (to avoid allocating a new tuple or vector), and have `promote_col_type` work with that. It should be as efficient, and it doesn't even need to be a generated function. If we really want an optimized version hardcoding the number of arguments, we should define `vcat(dfs::AbstractDataFrame...)` separately.

Reply:
@nalimilan will you push a PR along the lines you proposed? I have tested what @quinnj proposed and it was not bad (for sure not a performance bottleneck).